test/unit/GenericTest.php

```diff
@@ -33,6 +33,12 @@ public function testCanConnectToSendGridApi()
 
         $sg4 = new SendGrid(self::$apiKey, ['curl' => [CURLOPT_PROXY => '127.0.0.1:8000']]);
         $this->assertEquals($sg4->client->getCurlOptions(), [10004 => '127.0.0.1:8000']);
+
+        $subuser = 'abcxyz@this.is.a.test.subuser';
+        $headers[] = 'On-Behalf-Of: ' . $subuser;
+        $sg5 = new SendGrid(self::$apiKey, [], $subuser);
+        $this->assertEquals($sg5->client->getHeaders(), $headers);
```

Comment:
Regardless `assert*()` methods are already misused in the existing codebase, please, pass the expectation in argument 1, then the actual value in argument 2. [[ref](https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.assertEquals)].
Use `assertSame()` if possible.